core/domain/feature_flag_registry.py

```diff
@@ -0,0 +1,282 @@
+# coding: utf-8
+#
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Registry for feature flags."""
+
+from __future__ import annotations
+
+from core import platform_feature_list
+from core.domain import feature_flag_domain
+from core.platform import models
+
+from typing import Dict, List, Optional
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import config_models
+
+(config_models,) = models.Registry.import_models([models.Names.CONFIG])
+
+FeatureNames = platform_feature_list.FeatureNames
+
+
+class Registry:
+    """Registry for all feature flags."""
+
+    # The keys of feature_flag_spec_registry are the feature flags name,
+    # and the values are FeatureFlagSpec instances with initial settings
+    # defined in this file.
+    feature_flag_spec_registry: Dict[
+        str, feature_flag_domain.FeatureFlagSpec] = {}
+
+    @classmethod
+    def create_feature_flag(
```

Comment:
From [previous discussion](https://github.com/oppia/oppia/pull/18819#discussion_r1427998540):

> Sean: Also I don't understand why we have this method in the first place. Shouldn't we get the config values from the datastore and only if it doesn't exist, set it to the default values? Otherwise there's going to be confusion if this method is called for feature flags that already have values stored in the datastore. Why can't we instead always use get_feature_flag()?

> Hitesh: Actually, the `create_feature_flag` method gets used in the feature_flag_registry file and each time this file gets loaded we call the `create_feature_flag` method and this updates the feature_flag_spec_registry variable each time. The variable only stores FeatureFlagSpec for each feature flag, which is just the description and feature_stage. We do not want to make a call to the backend each time when the feature_flag_registry.py file gets loaded. I think this method will be required in order to update the feature_flag_spec_registry variable.
Thanks

I see ... but in that case, then why do we need the create_feature_flag() method at all and the registry at all? Why can't we just get the spec values on the fly when needed -- what operations are we saving by doing it the current way?

Reply:
Now that I look at the bigger picture I think this makes sense and we can avoid some of the complexities. I have mentioned issue #19439 regarding this and will start working upon it after this PR gets in.
Thanks

Reply:
I'm fine with punting the other thing in #19439 since it was already in the codebase, but this one I can't really allow in because it doesn't even make sense conceptually (we should not be creating "empty" feature flags and my original concern still applies). Can you remove it from this PR?

Reply:
Make sense, I have updated the codebase.
Thanks